net/net.go

```diff
@@ -16,7 +16,128 @@ limitations under the License.
 
 package net
 
-import "net"
+import (
+	"fmt"
+	"net"
+)
+
+// ParseCIDRs parses a list of cidrs and return error if any is in valid.
+// order is maintained
+func ParseCIDRs(cidrsString []string) ([]*net.IPNet, error) {
+	cidrs := make([]*net.IPNet, 0, len(cidrsString))
+	for _, cidrString := range cidrsString {
+		_, cidr, err := net.ParseCIDR(cidrString)
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse cidr value:%q with error:%v", cidrString, err)
+		}
+		cidrs = append(cidrs, cidr)
+	}
+	return cidrs, nil
+}
+
+// IsDualStackIPs returns if a slice of ips is:
+// - all are valid ips
+// - at least one ip from each family (v4 or v6)
+// - an error is returned if length is less than 2
+// - an error is returned if duplicate ips are found
+func IsDualStackIPs(ips []net.IP) (bool, error) {
+	if len(ips) < 2 {
+		return false, fmt.Errorf("ips length is less than 2")
+	}
+
+	v4Found := false
+	v6Found := false
+	seen := make(map[string]int)
```

Comment:
if you make this map[string]bool you can simply check seen[ip] - the default return for not-found will be false.

Reply:
same can be said for int. in both cases map value is not used. Because verbosity matter (same perf in both cases). validating if a key exist is has no gotchas and easier to read.

Reply:
yeah, but with bool I can say `if !seen[val] { seen[val] = true }` which reads really well :)

Minor point, if other changes being made

Reply:
this was removed, because duplicate check was removed.